core/src/main/scala/io/projectglow/common/GenomicIntervalUtils.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2019 The Glow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.projectglow.common
+
+import scala.math.{max, min}
+
+object GenomicIntervalUtils {
+
+  /**
+   * A class representing a closed interval of longs that also accommodates an empty interval.
+   */
+  class LongInterval() extends Serializable {
+    var interval: Option[(Long, Long)] = None
+
+    def this(start: Long, end: Long) {
+      this()
+      if (end >= start) {
+        this.interval = Option((start, end))
+      }
+    }
+
+    def start: Long =
```

Comment:
IMO it's bad practice for simple accessors like this to throw exceptions. It's better to just expose method that gets the optional start. I think that the simple `coordinates` covers that.